crates/api_common/src/utils.rs

```diff
@@ -928,10 +928,15 @@ async fn proxy_image_link_internal(
 
 /// Rewrite a link to go through `/api/v3/image_proxy` endpoint. This is only for remote urls and
 /// if image_proxy setting is enabled.
-pub async fn proxy_image_link(link: Url, context: &LemmyContext) -> LemmyResult<DbUrl> {
+pub async fn proxy_image_link(
+  link: Url,
+  is_thumbnail: bool,
+  context: &LemmyContext,
+) -> LemmyResult<DbUrl> {
```

Comment:
Only this function is used for post thumbnails, others like `proxy_image_link_api()` are for avatars etc which dont need to be downscaled.